powerscale/provider/namespace_acl_resource.go

```diff
@@ -0,0 +1,576 @@
+								},
+								"id": schema.StringAttribute{
+									Computed:            true,
+									Description:         "Specifies the serialized form of a persona, which can be 'UID:0' or 'GID:0'",
+									MarkdownDescription: "Specifies the serialized form of a persona, which can be 'UID:0' or 'GID:0'",
+								},
+								"name": schema.StringAttribute{
+									Computed:            true,
+									Description:         "Specifies the persona name, which must be combined with a type.",
+									MarkdownDescription: "Specifies the persona name, which must be combined with a type.",
+								},
+							},
+						},
+						"accessrights": schema.ListAttribute{
+							Computed:            true,
+							Description:         "Specifies the access control permissions for a specific user or group.",
+							MarkdownDescription: "Specifies the access control permissions for a specific user or group.",
+							ElementType:         types.StringType,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Configure configures the resource.
+func (r *NamespaceACLResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	powerscaleClient, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = powerscaleClient
+}
+
+// Create allocates the resource.
+func (r *NamespaceACLResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "creating namespace acl")
+
+	var plan models.NamespaceACLResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	namespaceACLToCreate := powerscale.NamespaceAcl{}
+	plan.ACL = plan.CustomACL
+	// Get param from tf input
+	err := helper.ReadFromState(ctx, plan, &namespaceACLToCreate)
+	if err != nil {
+		errStr := constants.CreateNamespaceACLErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error creating namespace acl",
+			fmt.Sprintf("Could not read namespace acl param with error: %s", message),
+		)
+		return
+	}
+	err = helper.CheckNamespaceACLParam(&namespaceACLToCreate)
+	if err != nil {
+		errStr := constants.CreateNamespaceACLErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error creating namespace acl",
+			fmt.Sprintf("Namespace acl param invalid with error: %s", message),
+		)
+		return
+	}
+
+	if namespaceACLToCreate.HasAcl() {
+		err = helper.UpdateNamespaceACL(ctx, r.client, plan, namespaceACLToCreate)
```

Comment:
better to add cases in test - leave acl_custom  empty or setting acl_custom = []

Reply:
And seems the resource is similar to settings, do you support `do import action when creating with empty config` and if support, better also to add one test case for it.

Reply:
Yes, cases added